feat(label): initial design guidance (#33)
vmware-clarity/clarity-guidance: 1015-label-design-guidance.md

---
category: component design guidance
state: draft
created: 2024-10-15
updated: 2024-10-21
---

# Label Design Guidance

> [Label](https://clarity.design/documentation/label) component show concise metadata in a compact format.

Reviewer: ```suggestion
The [label](https://clarity.design/documentation/label) component shows concise metadata in a compact format.
```

## Guidance

- **Must** keep labels short and concise.

> - **Must** NOT use all capital letters in labels. Using all capital letters might make the user think the label as a button and conflates the use case for labels with actions and buttons.

Reviewer: ```suggestion
- **Must not** use all capital letters in labels. Using all capital letters might make the user think the label is a button and conflates the use case for labels with actions and buttons.
```

> - **Must** NOT use multiple lines of text within a single label.

Reviewer: ```suggestion
- **Must not** use multiple lines of text within a single label.
```

- **Should** use parenthesis for any additional metadata, if required in the same label.
- **Should** use [clickable labels](https://clarity.design/documentation/label#clicking-labels) to perform an action related to that label.

> - **Should** use a close icon at the right-most side of a label to dismiss it.

Reviewer: ```suggestion
- **May** use a close icon at the right-most side of a label to dismiss it.
```

I think this one is better worded as a "may" because dismissing labels is optional. Not all labels "should" have a close button.

> - **Should** use labels and [badges](https://clarity.design/documentation/badge) together to show a count relating to the metadata displayed in the label.

Reviewer: If there is a count related to the label, is using a badge within the label *required* or *optional*. It it is required, this should be reworded to a must.

```suggestion
- If there is a count or other number related to the metadata displayed in the label, a badge **must** be used.
```

## Style and Color

- **Must** choose colors for the label based on [Clarity color palette](https://clarity.design/documentation/color).

> - **Must** avoid using too many colors within the same context. Displaying too many colors may distract the user from the core of your application and the information it presents.

Reviewer: ```suggestion
- **Should not** use too many colors within the same context. Displaying too many colors may distract the user from the core of your application and the information it presents.
- **Must not** use color alone to convey information.
```

I think it would be good to add a line for accessibility here.

---
category: component design guidance
state: draft
created: 2024-10-15
updated: 2024-10-22
---

# Label Design Guidance

>  [Label](https://clarity.design/documentation/label) component shows concise metadata in a compact format.

Reviewer 2: ```suggestion
 A [label](https://clarity.design/documentation/label) component provides a concise description or identifier for interface elements, like form fields or statuses, helping users quickly understand their purpose.
```

## Guidance

> - **Must** keep labels short and concise.

Reviewer 2: ```suggestion
- Design teams **must** keep labels short and concise to ensure they are easy to read and quickly understood.
```

> - **Must not** use all capital letters in labels. Using all capital letters might make the user think the label is a button and conflates the use case for labels with actions and buttons.

Reviewer 2: ```suggestion
- Design teams **must not** use all capital letters in labels. Using all caps can confuse users, making them think the label is a button or actionable element. This also impacts readability and accessibility, as screen readers may interpret all caps differently, and users with cognitive impairments may find it harder to read.
```

> - **Must not** use multiple lines of text within a single label.

Reviewer 2: ```suggestion
- Design teams **must not** use multiple lines of text within a single label to keep the interface clear and easy to scan.
```

> - **Should** use parenthesis for any additional metadata, if required in the same label.

Reviewer 2: ```suggestion
- Design teams **should** use parentheses for any additional metadata required within the same label to keep the information organized and easy to read.
```

> - **Should** use [clickable labels](https://clarity.design/documentation/label#clicking-labels) to perform an action related to that label.

Reviewer 2: ```suggestion
- Design teams **should** use [clickable labels](https://clarity.design/documentation/label#clicking-labels) to perform actions related to the label, providing a more intuitive user experience.
```

> - **May** use a close icon at the right-most side of a label to dismiss it unless the label itself is clickable.

Reviewer: ```suggestion
- **May** use a close icon at the right-most side of a label to dismiss it. **Must not** have a separate action for a dismissible label. The action for a dismissible label *is* to dismiss the label.
```

I just realized that all dismissible labels *are* clickable labels, so my original wording suggesting is a bit misleading. This might be better. Thoughts?

> - **May** use a close icon at the right-most side of a label to dismiss it. **Must not** have a separate action for a dismissible label because the action for a dismissible label *is* to dismiss the label.

Reviewer 2: ```suggestion
- Design teams **may** use a close icon at the right-most side of a label to dismiss it.  Design teams **must not** assign a separate action for a dismissible label, as its primary action is to dismiss the label.
```

> - If there is a count or other number related to the metadata displayed in the label, a badge **must** be used.

Reviewer 2: ```suggestion
- If a count or other number is related to the metadata displayed in the label, a badge **must** be used to clearly convey the information.
```

## Style and Color

- **Must** choose colors for the label based on [Clarity color palette](https://clarity.design/documentation/color).

> - **Should not** use too many colors within the same context. Displaying too many colors may distract the user from the core of your application and the information it presents.

Reviewer 2: ```suggestion
- Design teams **should not** use excessive colors within the same context. As a rule of thumb, aim to limit colors to only those necessary for conveying meaning and maintaining visual harmony. Overusing colors can distract users and create cognitive or accessibility challenges.
```

- **Must not** use color alone to convey information.

> - It is recommended to reserve red, yellow, and green to display state or status.

Reviewer: ```suggestion
- **Should** reserve red, yellow, and green to display state or status.
```

> - **Must not** use color alone to convey information.

Reviewer 2: ```suggestion
- Design teams **must not** use color alone to convey information, as this can make content inaccessible to users with visual impairments or color vision deficiencies. Use additional indicators, such as text or icons, to ensure all users can interpret the information.
```
